app/src/main/java/com/likeminds/feedsx/feed/view/FeedFragment.kt

```diff
@@ -237,6 +249,57 @@ class FeedFragment :
         }
     }
 
+    // processes delete post request
+    private fun deletePost(postId: String) {
+        //TODO: set isAdmin
+        val isAdmin = true
+        if (isAdmin) {
+            val deleteEntityExtras = DeleteEntityExtras.Builder()
+                .entityId(postId)
+                .entityType(DELETE_ENTITY_TYPE_POST)
+                .build()
+            DeleteEntityDialogFragment.showDialog(
+                childFragmentManager,
+                deleteEntityExtras
+            )
+        } else {
+            showDeletePostDialog()
+        }
+    }
+
+    // shows delete post dialog when user deletes their own post
+    private fun showDeletePostDialog() {
+        val builder = AlertDialog.Builder(requireContext())
+        builder.setMessage(getString(R.string.delete_post_message))
+            .setTitle(getString(R.string.delete_post_question))
+            .setCancelable(true)
+            .setPositiveButton(getString(R.string.delete_caps)) { _, _ ->
+
+            }.setNegativeButton(getString(R.string.cancel_caps)) { _, _ ->
+                alertDialog.dismiss()
+            }
+        //Creating dialog box
+        alertDialog = builder.create()
+        alertDialog.show()
+        alertDialog.getButton(AlertDialog.BUTTON_NEGATIVE)
+            ?.setTextColor(ContextCompat.getColor(requireContext(), R.color.black_40))
+    }
```

Comment:
dont create dialog using this method. create a new class using BaseDialogFragment and show it here